### PR TITLE
Domains: Improve all domains management performance

### PR DESCRIPTION
**Pull request by rafaelgallani**
Depends on D119262-code.
In an effort to improve the all-domains management page - and make it usable for users with a large number of domains, we're trying some options. 
This PR adds a "light" version of the page - loaded for these users with many domains (we set the threshold to 100). 
It does some optimizations: 
- For the light version:
  - Removes the `Status` column and the ellipsis menu (`Action` column);
  - No site-specific requests are performed (`GET /sites/*`);
  - No dependency on the `sites` data;
  - No dependency on the `purchases` data;
- For all versions:
  - Removes the "Auto-renew" toggle;
  - Doesn't use the `capabilities` state to show the manageable domains - these are now filtered on the backend;
  - Sets the initial sorting to the domain name, ascending;

### Light version
image

### Regular version (unchanged)
image

## Testing Instructions
- For an account with many domains (> 100), go to `/domains/manage`:
  - Ensure that the page loads - you won't get a timeout anymore
  - Ensure that the page doesn't perform a rerender for each domain - the page doesn't rely on the `sites` state anymore, so no requests to `/sites/*` are performed anymore, thus no rerenders.
  - Ensure that the navigation works and no errors (OOM, timeouts, etc.) happen.
  - Confirm that the notices, "Status" and "Actions" columns are not rendered;
- Ensure that the site domains component is unchanged - and working the same way as it was before;

## Pre-merge Checklist

- [X] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [X] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/domains/domain-management/list/domain-row.jsx [hunk 1]
```diff
@@ -98,22 +98,26 @@ class DomainRow extends PureComponent {
...
+		if ( domain.isDomainOnlySite ) {
+			return null;
+		}
+
 		return (
 			<div className="domain-row__site-cell">
-				<Button href={ '/home/' + site?.slug } plain>
-					{ ! site.options?.is_domain_only ? site?.title || site?.slug : '' }
+				<Button href={ '/home/' + domain.blogId } plain>
+					{ ( domain.siteTitle || domain.siteSlug ) ?? '' }
```
Comment (ramynasr) [inline client/my-sites/domains/domain-management/list/domain-row.jsx]: Is that the same thing?

```suggestion
					{ domain.siteTitle ?? domain.siteSlug ?? '' }
```

File: client/my-sites/domains/domain-management/list/domain-row.jsx [hunk 2]
```diff
@@ -126,7 +130,13 @@ class DomainRow extends PureComponent {
 	}
 
 	renderDomainStatus() {
-		const { currentRoute, domain, site, isLoadingDomainDetails, translate, dispatch } = this.props;
+		const { columns, currentRoute, domain, site, isLoadingDomainDetails, translate, dispatch } =
+			this.props;
```
Comment (ramynasr) [inline client/my-sites/domains/domain-management/list/domain-row.jsx]: Nit: Why the new line? Do we do hard-wraps?

File: client/my-sites/domains/domain-management/list/domain-row.jsx [hunk 3]
```diff
@@ -126,7 +130,13 @@ class DomainRow extends PureComponent {
...
+
+		if ( ! columns.find( ( column ) => column.name === 'status' ) ) {
```
Comment (ramynasr) [inline client/my-sites/domains/domain-management/list/domain-row.jsx]: Would that loop through all the columns for every domain row? Would checking for `IsLightVersion` to know if we need the status column or not be more efficient?

File: client/my-sites/domains/domain-management/list/domain-row.jsx [hunk 4]
```diff
@@ -361,8 +327,13 @@ class DomainRow extends PureComponent {
 			showDomainDetails,
 			site,
 			translate,
+			columns,
 		} = this.props;
 
+		if ( ! columns.find( ( column ) => column.name === 'action' ) ) {
```
Comment (ramynasr) [inline client/my-sites/domains/domain-management/list/domain-row.jsx]: Same as the comment above about using `isLightVersion`.

File: client/my-sites/domains/domain-management/list/all-domains.jsx [hunk 5]
```diff
@@ -756,29 +806,29 @@ const getFilteredDomainsList = ( state, context ) => {
 			} );
 
 		default:
-			return domainsList.filter(
-				( domain ) => domain.type !== domainTypes.WPCOM && canManageSitesMap[ domain.blogId ]
```
Comment (ramynasr) [inline client/my-sites/domains/domain-management/list/all-domains.jsx:760]: would removing `canManageSitesMap[ domain.blogId ]` mean that a user who is a member of a site but not an admin would be able to see a list of domains owned by other admins of the same site?

File: client/my-sites/domains/domain-management/list/domain-row.jsx (hunk 2 above)
Comment (rafaelgallani) [reply client/my-sites/domains/domain-management/list/domain-row.jsx]: Linter auto-fix. 😬 
image

File: client/my-sites/domains/domain-management/list/all-domains.jsx (hunk 5 above)
Comment (rafaelgallani) [reply client/my-sites/domains/domain-management/list/all-domains.jsx:760]: No, the domains are now filtered on the backend - the user should only see sites that they're able to manage.
See here: D119262-code (add this fragment to the URL `#change-cOHTK7Y8aug5`), line 100

**Pull request merged**